sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs

```diff
@@ -77,75 +83,51 @@ public ActivityTagsProcessor()
-                if (_currentActivityType == OperationType.Azure)
-                {
-                    HasAzureNamespace = true;
-
-                    // For Dependency telemetry, azure namespace set as type, so need to set in properties.
-                    // Hence adding to MappedTags.
-                    if (activity.GetTelemetryType() == TelemetryType.Dependency)
-                    {
-                        AzMonList.Add(ref MappedTags, tag);
-                    }
-                    else
+                    switch (tag.Key)
                     {
-                        // For request telemetry, azure namespace is not processed and should be included in the properties.
-                        AzMonList.Add(ref UnMappedTags, tag);
+                        case SemanticConventions.AttributeHttpMethod:
+                            activityType |= OperationType.Http;
+                            break;
+                        case SemanticConventions.AttributeDbSystem:
+                            activityType |= OperationType.Db;
+                            break;
+                        case SemanticConventions.AttributeMessagingSystem:
+                            activityType |= OperationType.Messaging;
+                            break;
+                        case SemanticConventions.AttributeAzureNameSpace:
+                            activityType |= OperationType.Azure;
+                            break;
                     }
-
-                    continue;
-                }
-
-                // If the current and previous activity types are the same or the previous activity type is unknown (initialized value),
-                // set activityType to _currentActivityType and add the tag to the MappedTags list.
-                // This ensures that we track semantics of the same type. For example, if HTTP and SQL semantic tags are mixed,
-                // whichever comes first in the tags list wins, and mapping happens for that type.
-                if (_currentActivityType == previousActivityType || previousActivityType == OperationType.Unknown)
-                {
-                    activityType = _currentActivityType;
-                    AzMonList.Add(ref MappedTags, tag);
                 }
                 else
                 {
+                    // If the tag value is an array, there is no need to check for semantics;
+                    // directly add it to the Unmapped list.
+                    if (tag.Value is Array array)
+                    {
+                        AzMonList.Add(ref UnMappedTags, new KeyValuePair<string, object?>(tag.Key, array.ToCommaDelimitedString()));
+                        continue;
+                    }
+
                     AzMonList.Add(ref UnMappedTags, tag);
                 }
+            }
 
-                previousActivityType = _currentActivityType;
+            if ((activityType & OperationType.Azure) == OperationType.Azure)
```

Comment:
```suggestion
            if ((activityType.HasFlag(OperationType.Azure)))
```

Reply:
I intentionally used this syntax, as I was removing `OperationType.Azure`, using this one - `activityType ^= OperationType.Azure;`.  If you think using HasFlag would improve readability, please let me know

Reply:
Not a blocker for me.
I think `HasFlag` is easier to read, but it's doing this same operation. :)
https://github.com/dotnet/runtime/blob/53af66a48215f0b7833dce499343eccae1dc9dff/src/libraries/System.Private.CoreLib/src/System/Enum.cs#L400

Reply:
After you point me to the code, I'm thinking why to use `HasFlag`. Current code will work better as it reduces few lines of code executing within `HasFlag` method. :blush:

Reply:
Removed this code to make it easier.